src/randgraphs.jl

```diff
@@ -107,9 +107,13 @@ function _suitable(edges::Set{Edge}, potential_edges::Dict{Int, Int})
     return false
 end
 
-function _try_creation(n::Int, k::Int)
+function _try_creation(n::Int,@compat k::Union{Int,Array{Int}})
```

Comment:
Yes, we can, if you really think this could impact performance. Maybe we can group the common part (most part) in a third function to avoid unnecessary duplication